### PR TITLE
feat: Prevent leaving the page without saving

### PR DESCRIPTION
**Pull request by liordino**
feat: created Prompt component to avoid leaving pages without saving

feat: applied component to the editable table

feat: applied component to the Cashout, Commissions, Locales, Cashboxes,
Notifications, CryptoBalanceOverrides and Wallet pages

feat: applied component to the ContactInfo and ReceiptPrinting pages

refactor: export the default prompt message to be used in other contexts

fix: applied prompt component to the Operator Info pages

fix: create routes for the operator info components

feat: applied the Prompt component to the Contact Info and Receipt pages

feat: applied the Prompt component to the Terms & Conditions page

### CODE REVIEW

**Comment by liordino** [comment]
[Prevent leaving the page without saving](https://trello.com/c/fqVxcOWq/189-prevent-leaving-the-page-without-saving)

File: new-lamassu-admin/src/components/editableTable/Table.js [hunk 1]
```diff
@@ -49,6 +50,9 @@ const ETable = ({
   disableRowEdit,
   createText = 'Add override'
 }) => {
+  console.log(data)
+  console.log(elements)
+
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/components/editableTable/Table.js]: Remove the console.logs

File: new-lamassu-admin/src/components/booleanPropertiesTable/BooleanPropertiesTable.js [hunk 2]
```diff
@@ -16,102 +20,95 @@ const useStyles = makeStyles(booleanPropertiesTableStyles)
...
+                      {!editing && (
+                        <FormikField
+                          component={BooleanCell}
+                          name={it.name}
+                          className={classes.rightTableCell}
+                        />
+                        // <BooleanCell
+                        //   className={classes.rightTableCell}
+                        //   value={it.value}
+                        // />
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/components/booleanPropertiesTable/BooleanPropertiesTable.js]: Remove comments

File: new-lamassu-admin/src/components/booleanPropertiesTable/BooleanPropertiesTable.js [hunk 3]
```diff
@@ -16,102 +20,95 @@ const useStyles = makeStyles(booleanPropertiesTableStyles)
...
+                            classes.rightTableCell
+                          )}
+                        />
+                      )}
+                      {!editing && (
+                        <FormikField
+                          component={BooleanCell}
+                          name={it.name}
+                          className={classes.rightTableCell}
+                        />
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/components/booleanPropertiesTable/BooleanPropertiesTable.js]: Move the BooleanCell component to within this file, also make it really simple like:

```js
(name) => {
  const { values } = useFormikContext()
  values[name] ? <TrueIcon/> : <FalseIcon/>
}
```

It doesn't need to be a formik field.

File: new-lamassu-admin/src/components/tables/BooleanCell.js [hunk 4]
```diff
@@ -3,6 +3,7 @@ import React from 'react'
 import { ReactComponent as FalseIcon } from 'src/styling/icons/table/false.svg'
 import { ReactComponent as TrueIcon } from 'src/styling/icons/table/true.svg'
 
-const BooleanCell = ({ value }) => (value ? <TrueIcon /> : <FalseIcon />)
+const BooleanCell = ({ name, value }) =>
+  value ? <TrueIcon name={name} /> : <FalseIcon name={name} />
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/components/tables/BooleanCell.js]: remove this component.

File: new-lamassu-admin/src/components/tables/formik/BooleanCell.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import React, { memo } from 'react'
+
+import BooleanCell from '../BooleanCell'
+
+const BooleanCellFormik = memo(({ label, ...props }) => {
+  const { name, value } = props.field
+
+  return <BooleanCell name={name} value={value === 'true'} {...props} />
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/components/tables/formik/BooleanCell.js]: remove this component.

File: new-lamassu-admin/src/components/tables/formik/index.js [hunk 6]
```diff
@@ -0,0 +1,3 @@
+import BooleanCell from './BooleanCell'
+
+export { BooleanCell }
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/components/tables/formik/index.js]: remove this file

**Review by RafaelTaranto — CHANGES_REQUESTED** [review]

**Review by RafaelTaranto — APPROVED** [review]

**Pull request merged**